engine/src/main/java/org/terasology/engine/entitySystem/event/internal/EventSystemImpl.java

```diff
@@ -95,11 +95,11 @@ public void registerEvent(ResourceUrn uri, Class<? extends Event> eventType) {
     public void registerEventHandler(ComponentSystem handler) {
         Class handlerClass = handler.getClass();
         if (!Modifier.isPublic(handlerClass.getModifiers())) {
-            logger.error("Cannot register handler {}, must be public", handlerClass.getName());
+            logger.atError().addArgument(() -> handlerClass.getName()).log("Cannot register handler {}, must be public");
             return;
         }
 
-        logger.debug("Registering event handler {}", handlerClass.getName());
+        logger.atDebug().addArgument(() -> handlerClass.getName()).log("Registering event handler {}");
```

Comment:
`getName()` is most likely O(1) and only called once there when registering event handlers, so impact can be considered neglectable
```suggestion
        logger.debug("Registering event handler {}", handlerClass.getName()); //NOPMD
```